testsuite/features/support/cobbler_test.rb

```diff
@@ -148,48 +156,23 @@ def system_create(name, profile)
   # Args:
   #   name: The name of the system to be removed.
   def system_remove(name)
-    raise "system cannot be found. #{$ERROR_INFO}" unless system_exists(name)
+    raise(IndexError, "System cannot be found. #{$ERROR_INFO}") unless element_exists('systems', name)
+
     begin
       @server.call('remove_system', name, @token)
-    rescue
-      raise "deleting system failed. #{$ERROR_INFO}"
+    rescue StandardError
+      raise(StandardError, "Deleting system failed. #{$ERROR_INFO}")
     end
   end
 
   ##
-  # Checks if a distribution exists in the database by using 'distro' as the table name, 'name' as the column and the name of the distro.
+  # Checks if a distribution exists in the database by using 'distro' as the table name,
+  # 'name' as the column and the name of the distro.
```

Comment:
```suggestion
  # Checks if a Cobbler item exists in the database by using 'distros|profiles|systems|repos' as the table name,
  # 'name' as the column and the name of the item.
```
This has to be adjusted since we are not only talking about a distro here.